nomenclature/cli.py

```diff
@@ -41,10 +43,34 @@ def cli_valid_yaml(path: Path):
     default="definitions",
 )
 def cli_valid_project(
-    path: Path, dimensions: List[str], mappings: str, definitions: str
+    path: Path,
+    dimensions: Optional[List[str]],
+    mappings: Optional[str],
+    definitions: str,
 ):
     """Assert that `path` is a valid project nomenclature
 
+    Parameters
+    ----------
+    path : Path
+        directory path to the file of interest
+    dimensions : List[str], optional
+        List of dimensions to be checked, default to None which implies that all
+        directories in `dimensions` will be checked
```

Comment:
Alternative, more concise formulation
```suggestion
        Dimensions to be checked, defaults to all sub-folders of `dimensions`
```